### PR TITLE
Added image upload handling

### PR DESCRIPTION
**Pull request by gohabereg**

### CODE REVIEW

File: src/resolvers/project.js [hunk 1]
```diff
@@ -32,21 +34,32 @@ module.exports = {
      * @param {Context.user} user - current authorized user {@see ../index.js}
      * @return {Project[]}
      */
-    async createProject(_obj, { workspaceId, name }, { user }) {
+    async createProject(_obj, { workspaceId, name, image }, { user }) {
```
Comment (neSpecc) [inline src/resolvers/project.js]: jsdoc is missed

File: src/resolvers/project.js [hunk 2]
```diff
@@ -32,21 +34,32 @@ module.exports = {
...
+        workspaceId,
       ]);
 
       if (!workspace) {
         throw new ValidationError('No such workspace');
       }
 
-      const project = await Project.create({
+      if (image) {
+        image = await image;
```
Comment (neSpecc) [inline src/resolvers/project.js]: if `image` is a promise, to better to change its naming

File: src/resolvers/user.ts [hunk 3]
```diff
@@ -158,7 +159,7 @@ export default {
      */
     async updateProfile(
       _obj: undefined,
-      { name, email }: {name: string; email: string},
+      { name, email, image }: {name: string; email: string, image: any},
```
Comment (neSpecc) [inline src/resolvers/user.ts]: why `any`?

Comment (neSpecc) [reply src/resolvers/user.ts]: jdoc is missed

File: src/resolvers/workspace.js [hunk 4]
```diff
@@ -187,19 +206,37 @@ module.exports = {
      * @param {Context.user} user - current authorized user
      * @returns {Promise<Boolean>}
      */
-    async updateWorkspace(_obj, { id, name, description }, { user }) {
+    async updateWorkspace(_obj, { id, name, description, image }, { user }) {
```
Comment (neSpecc) [inline src/resolvers/workspace.js]: jsdoc is missed

**Comment by neSpecc** [comment]
image

File: src/utils/files.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Save file to uploads dir
+ *
+ * @param {Readable} file - ReadStream with file contents
+ * @param {string} mimetype - file mimetype
+ *
+ * @return {string} - url to saved file
+ */
+export const save = (file: Readable, mimetype: string): string => {
```
Comment (nikmel2803) [inline src/utils/files.ts]: уБеДиТеЛьНаЯ просьба, давай функции будут функциями :D
https://hudochenkov.com/posts/prefer-function-declarations/

File: src/resolvers/workspace.js [hunk 6]
```diff
@@ -188,16 +202,19 @@ module.exports = {
      * @param {string} workspaceId - id of the updated workspace
      * @param {string} name - workspace name
      * @param {string} description - workspace description
+     * @param {Promise<FileUpload>} - workspace logo
+     * @param {Context.user} user - current authorized user
```
Comment (nikmel2803) [inline src/resolvers/workspace.js]: лишняя дока (внизу есть)

File: src/resolvers/user.ts [hunk 7]
```diff
@@ -154,11 +156,12 @@ export default {
      * @param name - user's name to change
      * @param email - user's email to change
      * @param user - current authenticated user
+     * @param {Promise<FileUpload>}image - user avatar
```
Comment (nikmel2803) [inline src/resolvers/user.ts]:
```suggestion
     * @param {Promise<FileUpload>} image - user avatar
```

File: src/resolvers/project.js [hunk 8]
```diff
@@ -28,11 +30,13 @@ module.exports = {
      * @param {ResolverObj} _obj
      * @param {string} workspaceId - workspace ID
      * @param {string} name - project name
+     * @param {Promise<FileUpload>} image - project logo
+     * @param {Context.user} user - current authorized user {@see ../index.js}
```
Comment (nikmel2803) [inline src/resolvers/project.js]: тоже лишняя дока

File: src/resolvers/workspace.js [hunk 9]
```diff
@@ -58,13 +60,25 @@ module.exports = {
          *   name: defaultPlan.name
          * };
          */
-        const workspace = await Workspace.create({
+
+        if (image) {
+          image = await image;
```
Comment (neSpecc) [inline src/resolvers/workspace.js]: `await upload` looks better

**Pull request merged**